artifactory_cleanup/rules/delete.py

```diff
@@ -92,37 +93,85 @@ class delete_empty_folder(Rule):
     """
 
     def _aql_add_filter(self, aql_query_list):
-        update_dict = {
-            "repo": {
-                "$match": "deleteEmptyFolder",
-            }
-        }
-        aql_query_list.append(update_dict)
+        # Get list of all files and folders
+        all_files_dict = {"path": {"$match": "**"}, "type": {"$eq": "any"}}
+        aql_query_list.append(all_files_dict)
         return aql_query_list
 
     def _filter_result(self, result_artifact):
-        r = self.artifactory_session.get(
-            "{}/api/repositories?type=local".format(self.artifactory_server)
-        )
-        r.raise_for_status()
-        repositories = r.json()
-
-        for count, repository in enumerate(repositories, start=1):
-            if repository["packageType"] == "GitLfs":
-                # GitLfs should be handled by the jfrog cli: https://jfrog.com/blog/clean-up-your-git-lfs-repositories-with-jfrog-cli/
-                print(
-                    f"Skipping '{repository['key']}' because it is a Git LFS repository"
-                )
-                continue
-
-            url = "{}/api/plugins/execute/deleteEmptyDirsPlugin?params=paths={}".format(
-                self.artifactory_server, repository["key"]
-            )
-
-            print(
-                f"Deleting empty folders for '{repository['key']}' - {count} of {len(repositories)}"
-            )
-            r = self.artifactory_session.post(url)
-            r.raise_for_status()
-
-        return []
+
+        # convert list of files to dict
+        # Source: https://stackoverflow.com/a/58917078
+
+        def nested_dict():
```

Comment:
Inline functions are not readable here, it's hard to get the idea from it. Could we move as much as possible outside of the class? 
I see there's no `self` usages here

Reply:
I will move this to a new utils.py file. It's more readable then.
I would still keep it as inline functions since this is a function very specific to the conversion of list to dict, and not used anywhere else.

Reply:
It might be either a separate function or class method, but differently not a function inside class method

Reply:
Did some refactoring now. From my point of view it's now ready to be merged.

I would still argue against putting this as a separate function. It would be out of context. If you still disagree, feel free to direclty update this MR.